src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java

```diff
@@ -22,127 +22,135 @@
     public String courseId;
     public String name;
     public String email;
-    public Boolean isArchived;
     public String key;
     public String role;
-    public boolean isDisplayedToStudents;
     public String displayedName;
-
+    public Boolean isArchived;
+    public boolean isDisplayedToStudents;
     public InstructorPrivileges privileges;
 
     /**
-     * Creates a new instructor with default access level and default displayedName.
-     *
-     * @deprecated only to be used for testing
+     * Builder class to simplify creation of object.
      */
-    @Deprecated
-    public InstructorAttributes(String googleId, String courseId, String name, String email) {
-        this(googleId, courseId, name, email,
-             Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER, DEFAULT_DISPLAY_NAME,
-             new InstructorPrivileges(Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER));
-    }
+    public static class Builder {
```

Comment:
A quick review. I'll give detailed review after the changes made.
1. Builder Pattern is used especially when you have optional fields. Could you please identify the optionals fields and require fields and group them together with some comments?

2. Also, put the Builder inner class at the end of the `Instructor` class

3. Let's say we use `with*` method name for optional fields (e.g. `displayedName()` -> `withDisplayedName()`)